linfa-kernel/src/lib.rs

```diff
@@ -82,13 +84,33 @@ impl<'a, F: Float> Kernel<ArrayView2<'a, F>> {
         }
     }
 
+    /// Performs the element-wise product between the kernel matrix
```

Comment:
this performs a matrix vector and not the hadamard product

Reply:
so `mul` is just like `dot`? Isn't `rhs` a matrix too?  I think I got confused because for the linear inner product `mul` and then `sum` is used

Reply:
Or does the resulting matrix have in each column the matrix vector product of the kernel matrix by the corresponding sample?

Reply:
I quick checked what `mul` actually does and its behaviour is the dot product for sparse matrices (`sprs`) and the hadamard product for dense matrices (`ndarray`). I did not catch the error because all tests in `linfa-reduction` uses sparse problems :/ please change that to `dot` for the dense case

Reply:
Whew! Will do 👍🏻